docs/src/pages/components/data-grid/filtering/ColumnTypeFilteringGrid.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import {
+  DataGrid,
+  ColTypeDef,
+  getNumericColumnOperators,
+  PreferencePanelsValue,
+} from '@material-ui/data-grid';
+import { useDemoData } from '@material-ui/x-grid-data-generator';
+
+const priceColumnType: ColTypeDef = {
+  extendType: 'number',
+  filterOperators: getNumericColumnOperators()
+    .filter((operator) => operator.value === '>' || operator.value === '<')
+    .map((operator) => {
+      return {
+        ...operator,
+        InputComponentProps: {
+          InputProps: {
+            startAdornment: <InputAdornment position="start">$</InputAdornment>,
+          },
+        },
+      };
+    }),
+};
+
+export default function ColumnTypeFilteringGrid() {
+  const { data } = useDemoData({ dataSet: 'Commodity', rowLength: 100 });
+  const [cols, setCols] = React.useState(data.columns);
+
+  React.useEffect(() => {
+    if (data.columns.length > 0) {
+      const visibleFields = ['desk', 'commodity', 'totalPrice'];
+      const demoCols = data.columns.map((col) => {
+        const newCol = { ...col, hide: visibleFields.indexOf(col.field) === -1 };
+        if (newCol.field === 'totalPrice') {
+          newCol.type = 'price';
+        }
+        return newCol;
+      });
+      setCols(demoCols);
+    }
+  }, [data]);
+
+  return (
+    <div style={{ height: 400, width: '100%' }}>
+      <DataGrid
+        rows={data.rows}
+        columns={cols}
```

Comment:
I think that we should ban shortcuts, at least for the documentation :) 

```suggestion
        columns={columns}
```